lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -2104,6 +2122,27 @@ bool SentNeverSendableDiagnosticInferrer::initForIsolatedPartialApply(
   // going to be returning some form of a SILFunctionArgument which is always
   // easy to find a name for.
   if (auto rootValueAndName = inferNameAndRootHelper(op->get())) {
+    // See if our underlying value is a SILBox type that contains a Sendable
+    // type. In that case, we could have only emitted this error if we were
+    // escaping the box itself. In such a case, we want to emit a better
+    // diagnostic that mentions that the reason we are emitting an error is b/c
+    // the value is mutable.
+    if (auto boxTy = op->get()->getType().getAs<SILBoxType>();
+        boxTy && boxTy->getLayout()->isMutable() &&
+        llvm::all_of(boxTy->getLayout()->getFields(),
+                     [&op](const SILField &field) -> bool {
+                       auto fieldTy = field.getAddressType();
+                       if (fieldTy.hasTypeParameter())
+                         fieldTy =
+                             op->getFunction()->mapTypeIntoEnvironment(fieldTy);
```

Comment:
I think you can use `SILType::isSendable(SILFunction *)` here instead, that I believe you added ;)